src/rating/rating.js

```diff
@@ -58,6 +58,12 @@ angular.module('ui.bootstrap.rating', [])
   this.render = function() {
     $scope.value = ngModelCtrl.$viewValue;
   };
+
+  $scope.$watch('value', function(value) {
```

Comment:
This seems to be an unnecessary & unperformant approach.

A better approach would be to create a formatter, that way it only gets run when the model value changes.